[2024-02-27T12:04:30.538+0700] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: missed_deadline_job_dag.spark_transform_missed_deadline_job manual__2024-02-27T04:28:02.034457+00:00 [queued]>
[2024-02-27T12:04:30.546+0700] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: missed_deadline_job_dag.spark_transform_missed_deadline_job manual__2024-02-27T04:28:02.034457+00:00 [queued]>
[2024-02-27T12:04:30.546+0700] {taskinstance.py:2170} INFO - Starting attempt 3 of 3
[2024-02-27T12:04:30.559+0700] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): spark_transform_missed_deadline_job> on 2024-02-27 04:28:02.034457+00:00
[2024-02-27T12:04:30.560+0700] {standard_task_runner.py:60} INFO - Started process 28446 to run task
[2024-02-27T12:04:30.564+0700] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'missed_deadline_job_dag', 'spark_transform_missed_deadline_job', 'manual__2024-02-27T04:28:02.034457+00:00', '--job-id', '125', '--raw', '--subdir', 'DAGS_FOLDER/missed_deadlinee_dag.py', '--cfg-path', '/tmp/tmphwhm_ekb']
[2024-02-27T12:04:30.565+0700] {standard_task_runner.py:88} INFO - Job 125: Subtask spark_transform_missed_deadline_job
[2024-02-27T12:04:30.652+0700] {task_command.py:423} INFO - Running <TaskInstance: missed_deadline_job_dag.spark_transform_missed_deadline_job manual__2024-02-27T04:28:02.034457+00:00 [running]> on host MinhNhut.
[2024-02-27T12:04:30.735+0700] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='missed_deadline_job_dag' AIRFLOW_CTX_TASK_ID='spark_transform_missed_deadline_job' AIRFLOW_CTX_EXECUTION_DATE='2024-02-27T04:28:02.034457+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-02-27T04:28:02.034457+00:00'
[2024-02-27T12:04:30.736+0700] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2024-02-27T12:04:30.736+0700] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'python /home/nhut/etl-***/pipeline/spark_missed_deadline_job.py']
[2024-02-27T12:04:30.740+0700] {subprocess.py:86} INFO - Output:
[2024-02-27T12:04:32.289+0700] {subprocess.py:93} INFO - 24/02/27 12:04:32 WARN Utils: Your hostname, MinhNhut resolves to a loopback address: 127.0.1.1; using 172.28.89.67 instead (on interface eth0)
[2024-02-27T12:04:32.292+0700] {subprocess.py:93} INFO - 24/02/27 12:04:32 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2024-02-27T12:04:42.511+0700] {subprocess.py:93} INFO - :: loading settings :: url = jar:file:/home/nhut/etl-***/.venv/lib/python3.10/site-packages/pyspark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2024-02-27T12:04:42.580+0700] {subprocess.py:93} INFO - Ivy Default Cache set to: /home/nhut/.ivy2/cache
[2024-02-27T12:04:42.580+0700] {subprocess.py:93} INFO - The jars for the packages stored in: /home/nhut/.ivy2/jars
[2024-02-27T12:04:42.584+0700] {subprocess.py:93} INFO - org.apache.hadoop#hadoop-aws added as a dependency
[2024-02-27T12:04:42.585+0700] {subprocess.py:93} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-c827f1b6-50bd-4fdf-90ac-3aa84a8a9d6d;1.0
[2024-02-27T12:04:42.585+0700] {subprocess.py:93} INFO - 	confs: [default]
[2024-02-27T12:04:42.711+0700] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-aws;3.3.2 in spark-list
[2024-02-27T12:04:42.739+0700] {subprocess.py:93} INFO - 	found com.amazonaws#aws-java-sdk-bundle;1.11.1026 in spark-list
[2024-02-27T12:04:42.766+0700] {subprocess.py:93} INFO - 	found org.wildfly.openssl#wildfly-openssl;1.0.7.Final in central
[2024-02-27T12:04:42.788+0700] {subprocess.py:93} INFO - :: resolution report :: resolve 194ms :: artifacts dl 9ms
[2024-02-27T12:04:42.788+0700] {subprocess.py:93} INFO - 	:: modules in use:
[2024-02-27T12:04:42.789+0700] {subprocess.py:93} INFO - 	com.amazonaws#aws-java-sdk-bundle;1.11.1026 from spark-list in [default]
[2024-02-27T12:04:42.789+0700] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-aws;3.3.2 from spark-list in [default]
[2024-02-27T12:04:42.789+0700] {subprocess.py:93} INFO - 	org.wildfly.openssl#wildfly-openssl;1.0.7.Final from central in [default]
[2024-02-27T12:04:42.789+0700] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2024-02-27T12:04:42.790+0700] {subprocess.py:93} INFO - 	|                  |            modules            ||   artifacts   |
[2024-02-27T12:04:42.790+0700] {subprocess.py:93} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2024-02-27T12:04:42.790+0700] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2024-02-27T12:04:42.790+0700] {subprocess.py:93} INFO - 	|      default     |   3   |   0   |   0   |   0   ||   3   |   0   |
[2024-02-27T12:04:42.790+0700] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2024-02-27T12:04:42.793+0700] {subprocess.py:93} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-c827f1b6-50bd-4fdf-90ac-3aa84a8a9d6d
[2024-02-27T12:04:42.793+0700] {subprocess.py:93} INFO - 	confs: [default]
[2024-02-27T12:04:42.800+0700] {subprocess.py:93} INFO - 	0 artifacts copied, 3 already retrieved (0kB/6ms)
[2024-02-27T12:04:43.018+0700] {subprocess.py:93} INFO - 24/02/27 12:04:43 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-02-27T12:04:43.314+0700] {subprocess.py:93} INFO - Setting default log level to "WARN".
[2024-02-27T12:04:43.314+0700] {subprocess.py:93} INFO - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2024-02-27T12:04:46.801+0700] {subprocess.py:93} INFO - 24/02/27 12:04:46 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2024-02-27T12:04:50.813+0700] {local_task_job_runner.py:302} WARNING - State of this instance has been externally set to restarting. Terminating instance.
[2024-02-27T12:04:50.816+0700] {process_utils.py:131} INFO - Sending 15 to group 28446. PIDs of all processes in the group: [28454, 28455, 28446]
[2024-02-27T12:04:50.816+0700] {process_utils.py:86} INFO - Sending the signal 15 to group 28446
[2024-02-27T12:04:50.816+0700] {taskinstance.py:2450} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-02-27T12:04:50.816+0700] {subprocess.py:104} INFO - Sending SIGTERM signal to process group
[2024-02-27T12:04:50.835+0700] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/nhut/etl-airflow/.venv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/nhut/etl-airflow/.venv/lib/python3.10/site-packages/airflow/operators/bash.py", line 203, in execute
    result = self.subprocess_hook.run_command(
  File "/home/nhut/etl-airflow/.venv/lib/python3.10/site-packages/airflow/hooks/subprocess.py", line 91, in run_command
    for raw_line in iter(self.sub_process.stdout.readline, b""):
  File "/home/nhut/etl-airflow/.venv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 2452, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2024-02-27T12:04:50.838+0700] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=missed_deadline_job_dag, task_id=spark_transform_missed_deadline_job, execution_date=20240227T042802, start_date=20240227T050430, end_date=20240227T050450
[2024-02-27T12:04:50.856+0700] {standard_task_runner.py:107} ERROR - Failed to execute job 125 for task spark_transform_missed_deadline_job (Task received SIGTERM signal; 28446)
[2024-02-27T12:04:51.150+0700] {process_utils.py:79} INFO - Process psutil.Process(pid=28454, status='terminated', started='12:04:30') (28454) terminated with exit code None
[2024-02-27T12:04:51.150+0700] {process_utils.py:79} INFO - Process psutil.Process(pid=28446, status='terminated', exitcode=1, started='12:04:30') (28446) terminated with exit code 1
[2024-02-27T12:04:51.323+0700] {process_utils.py:79} INFO - Process psutil.Process(pid=28455, status='terminated', started='12:04:30') (28455) terminated with exit code None
